src/LinearSolvers/LinearSolvers.jl

```diff
@@ -137,8 +136,8 @@ function ldiv!(::DirectSolver{Nothing},
 end
 get_transpose(::DirectSolver, M::CUSPARSE.CuSparseMatrixCSR) = CUSPARSE.CuSparseMatrixCSC(M)
 
-function update!(solver::AbstractIterativeLinearSolver, J)
-    update(J, solver.precond)
+function precond_update!(solver::AbstractIterativeLinearSolver, J, device)
```

Comment:
I am not sure about the naming of this function. I will keep `update!` (or at least rename it `update_preconditioner!` for consistency). 
Also: do we really need to pass the device here? I guess the device could be inferred by dispatching on the type of `J`

Reply:
I found the `update` functions confusing. In addition, what is updated is not even in the signature (linear solver vs preconditioner). I would vote for `update_preconditioner!` then.

Reply:
sounds good to me!